File: src/test/java/com/uapp_llc/test/stub/repository/identification/IdentificationContext.java
package com.uapp_llc.test.stub.repository.identification;

public class IdentificationContext<T> implements Identification<T> {

  private Identification<T> strategy;

  public void setStrategy(Identification<T> strategy) {
    this.strategy = strategy;
  }

  @Override
  public void apply(T entity) {
    strategy.apply(entity);
  }

}

File: src/test/kotlin/com/uapp_llc/test/stub/repository/identification/IdentificationContext.kt
package com.uapp_llc.test.stub.repository.identification

class IdentificationContext<T>(

    private var strategy: Identification<T>? = null

) : Identification<T> {

  fun setStrategy(strategy: Identification<T>) {
    this.strategy = strategy
  }

  override fun apply(entity: T) = strategy!!.apply(entity)

}
